marlowe-playground-client/src/MainFrame/State.purs

```diff
@@ -324,20 +353,34 @@ handleAction s (JavascriptAction action) = do
 handleAction settings (SimulationAction action) = do
   toSimulation (Simulation.handleAction settings action)
   case action of
-    ST.SetBlocklyCode -> do
-      mSource <- Simulation.editorGetValue
-      for_ mSource \source -> void $ query _blocklySlot unit (Blockly.SetCode source unit)
-      selectView BlocklyEditor
-    ST.EditHaskell -> selectView HaskellEditor
-    ST.EditJavascript -> selectView JSEditor
-    ST.EditActus -> selectView ActusBlocklyEditor
-    -- Replicate the state of unsavedChanges from the submodule/subcomponent into the MainFrame state
-    (ST.HandleEditorMessage (Monaco.TextChanged _)) -> do
-      hasUnsavedChanges <- queryCurrentEditorForUnsavedChanges
-      assign _hasUnsavedChanges hasUnsavedChanges
+    -- TODO: reimplement once we have the simulator in blockly and marlowe
+    ST.ViewAsBlockly -> pure unit
+    ST.EditSource -> do
+      flow <- use _workflow
+      case flow of
+        (Just lang) -> selectView $ selectLanguageView lang
+        _ -> pure unit
     _ -> pure unit
 
-handleAction _ SendBlocklyToSimulator = void $ query _blocklySlot unit (Blockly.GetCodeQuery unit)
+handleAction settings (BlocklyEditorAction action) = case action of
+  BL.SendToSimulator -> do
+    -- TODO: We used to block moving code to the simulation however due to the new UX there is no way to navigate this
+    -- I am leaving the old code here as we want to come up with a solution asap and it will involve using this same logic
+    -- The same occurs with Actus too
+    -- hasStarted <- use (_simulationState <<< _marloweState <<< to (\states -> (NEL.length states) > 1))
+    -- if hasStarted then
+    --   void $ query _blocklySlot unit (Blockly.SetError "You can't send new code to a running simulation. Please go to the Simulation tab and click \"reset\" first" unit)
+    -- else do
```

Comment:
I don't think this should be a problem as we don't intend the simulation to change the blockly code any more. When we do add the simulation pane to blockly we will want to do something specific with the blockly view anyway.

Reply:
Agreed, I just preserved an old comment, but it doesn't make sense any more.